src/llmcompressor/modeling/prepare.py

```diff
@@ -2,19 +2,58 @@
 from transformers import PreTrainedModel
 
 from llmcompressor.modeling.deepseek_v3 import replace as replace_DeepseekV3MoE
+from llmcompressor.modeling.qwen3_moe import replace as replace_Qwen3MoE
+from llmcompressor.utils.helpers import patch_attr
 
-__all__ = ["prepare_for_calibration"]
+__all__ = ["replace_modules_for_calibration"]
 
+# ---------------------- module replacements; permanent -------------------------
 replacements = {
     "DeepseekV3MoE": replace_DeepseekV3MoE,
 }
 
 
-def prepare_for_calibration(model: PreTrainedModel) -> PreTrainedModel:
+def replace_modules_for_calibration(model: PreTrainedModel) -> PreTrainedModel:
     for name, module in model.named_modules():
         cls_name = module.__class__.__name__
         if cls_name in replacements:
             new_module = replacements[cls_name](module)
             replace_module(model, name, new_module)
 
     return model
+
+
+# ------------------- module replacements; during calibration --------------------
+
+
+def update_qwen3_moe(model, stack):
+    for _, module in model.named_modules():
+        cls_name = module.__class__.__name__
+        if cls_name == "Qwen3MoeDecoderLayer":
```

Comment:
Could you use something like this pattern for matching? This way things don't break if the parent's structure changes, and we can also share matching logic between replacements
```
    for name, module in model.named_modules():
        cls_name = module.__class__.__name__
        if cls_name in replacements:
            new_module = replacements[cls_name](module)
            replace_module(model, name, new_module)
```